src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,157 @@
+from typing import Any, Union
+
+import fsspec
+
+from unstract.sdk.exceptions import FileOperationError
+from unstract.sdk.file_storage.constants import Common, FileSeekPosition
+from unstract.sdk.file_storage.fs_interface import FileStorageInterface
+from unstract.sdk.file_storage.helper import FileStorageHelper
+
+
+class FileStorage(FileStorageInterface):
+    # This class integrates fsspec library for file operations
+
+    fs: fsspec  # fsspec file system handle
+
+    def __init__(self, provider, credentials: dict[str, Any] = {}):
+        self.fs = FileStorageHelper.file_storage_init(
+            provider=provider, credentials=credentials
+        )
+
+    def read_file(
```

Comment:
@gaya3-zipstack We call this `read` itself.

Reply:
Done.